domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -245,72 +243,74 @@ class TopicListController @Inject constructor(
     topicProgressList: List<TopicProgress>,
     completionTimeFilter: (Long) -> Boolean
   ): List<PromotedStory> {
+    if (topicProgressList.isNotEmpty()) {
+      val playedPromotedStoryList = mutableListOf<PromotedStory>()
+      val sortedTopicProgressList =
+        topicProgressList.sortedByDescending {
+          val topicProgressStories = it.storyProgressMap.values
+          val topicProgressChapters = topicProgressStories.flatMap { it.chapterProgressMap.values }
```

Comment:
In this and above lines `it` can be confusing because of nesting.
Suggest using `topicProgress` and `storyProgress` respectively.

Reply:
Done,